### PR TITLE
Add transaction simulation test

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Add tests for transaction simulation for both ED25519 and Secp256k1
* single signer transaction with script / entry function / multisig payloads
* multi agent transaction with script / entry function payloads
* fee payer transaction with script / entry function / multisig payloads and as multi agent

### Test Plan
pnpm test

### CODE REVIEW

**Review by gregnazario — APPROVED** [review]

**Review by xbtmatt — APPROVED** [review]

File: src/transactions/transaction_builder/transaction_builder.ts [hunk 1]
```diff
@@ -328,17 +317,34 @@ export function generateSignedTransactionForSimulation(args: SimulateTransaction
...
+      accountAuthenticator.public_key,
+      accountAuthenticator.signature,
+    );
+  } else if (accountAuthenticator instanceof AccountAuthenticatorSecp256k1) {
+    transactionAuthenticator = new TransactionAuthenticatorSecp256k1(
+      accountAuthenticator.public_key,
+      accountAuthenticator.signature,
+    );
+  } else {
+    throw new Error("Invalud public key");
```
Comment (xbtmatt) [inline src/transactions/transaction_builder/transaction_builder.ts:332]:
```suggestion
    throw new Error("Invalid public key");
```

File: tests/e2e/transaction/transaction_simulation.test.ts [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+          transaction: rawTxn,
+        });
+        expect(response.success).toBeTruthy();
+      });
+      test("with multisig payload", async () => {
+        const rawTxn = await aptos.generateTransaction({
+          sender: senderAccount.accountAddress.toString(),
+          data: {
+            multisigAddress: secondarySignerAccount.accountAddress,
+            function: `0x${senderAccount.accountAddress.toStringWithoutPrefix()}::transfer::transfer`,
```
Comment (gregnazario) [inline tests/e2e/transaction/transaction_simulation.test.ts:75]: We should use `toString()` instead of `0x${toStringWithoutPrefix()}`

Comment (0xmaayan) [reply tests/e2e/transaction/transaction_simulation.test.ts:75]: it is a typing thing, `function` expects `0x${string}::${string}::${string}`, and using `toString()` typescript complains type mismatch

Comment (gregnazario) [reply tests/e2e/transaction/transaction_simulation.test.ts:75]: Oh, that's super annoying... we should probably fix that.

**Pull request merged**